src/internal/types.ts

```diff
@@ -96,15 +97,25 @@ type IsBoundedString<T> = T extends string
  *
  * @see EnumerableStringKeyedValueOf
  */
-export type EnumerableStringKeyOf<T> = `${Exclude<keyof T, symbol>}`;
+export type EnumerableStringKeyOf<T> =
+  T extends Record<infer K, unknown> ? `${Exclude<K, symbol>}` : never;
```

Comment:
**suggestion:** let's add tests for this and `EnumerableStringKeyedValueOf` to `types.test.ts`

testing would be wide cases (`Record<string, string>`, `Record<string | number, string>`, etc.), literals (`{ a: 1 }`, etc.), union types (like we have here), things with `symbol` (which is the intent behind this type in the first place), things with optional properties

Reply:
I agree, this would be very useful. I will add them in `types.test.ts` within the next few days.